### DISCUSSION TITLE
pac analytics - Set up the export of data to App Insights from the Power Platform Admin Center using PAC CLI

### DISCUSSION THREAD

**Discussion by jbuenoliveira** [microsoft/powerplatform-build-tools#1050] [Ideas]
### Context

The Power Platform Admin Center currently allows data export from Dataverse, Power Automate, Power Apps (preview), and Dynamics Customer Service (preview) to Application Insights. However, setting this up via CLI is currently not possible.
![image](https://github.com/user-attachments/assets/a124becd-67bc-43af-8404-7ed6f0b8c280)

Read more: [Export data to Application Insights](https://learn.microsoft.com/en-us/power-platform/admin/set-up-export-application-insights)

### Idea
It would be ideal if we could create and delete data export configurations for App Insights using PAC CLI. 
This feature would greatly enhance the automation of environment provisioning.

#### pac analytics

Usage: `pac analytics [export-app-insights] [export-data-lake]`

|command|description|
|--|--|
|`export-app-insights`|Manage data exports to Application Insights.|
|`export-data-lake`|(Preview) Manage data exports to Azure Data Lake.|

> `export-data-lake` sub-command is not relevant to this discussion, but we can explore it later as an extension of the `pac analytics` capabilities.

--------------------------------

#### pac analytics export-app-insights

Usage: `pac analytics export-app-insights [list] [create] [delete] [show]`

|command|description|
|--|--|
|`list`|List all data exports to Azure Application Insights.|
|`create`|Create a new data export to Azure Application Insights.|
|`delete`|Delete an existing data export to Azure Application Insights.|
|`show`|Displays information about a specific data export setup.|

--------------------------------

#### pac analytics export-app-insights list

Usage: `pac analytics export-app-insights list [--json]`

|command|description|
|--|--|
|`--json`|Returns the output of the command as a JSON formatted string.|

##### API Details

API URL: `[GET] https://emea.csanalytics.powerplatform.microsoft.com/api/v2/connections`

--------------------------------

#### pac analytics export-app-insights create

Usage: `pac analytics export-app-insights create [--package-name] [--environment] [--subscription-id] [--resource-group-game] [--resource-name] [--settings-file]`

|command|description|
|--|--|
|`--package-name`|Name of the data export. (alias: -pn)|
|`--environment`|Specifies the target Dataverse. The value may be a Guid or absolute https URL. When not specified, the active organization selected for the current auth profile will be used. (alias: -env)|
|`--subscription-id`|Subscription id. (alias: -sb)|
|`--resource-group-game`|Resource group name. (alias: -rg)|
|`--resource-name`|Application Insights resource name. (alias: -rn)|
|`--settings-file`|The .json file with the data export package settings. (alias: -sf)|

##### Settings File (*.json)

Dataverse diagnostics and performance:
```json
{
  "resourceProvider": "dataverse",
  "scenarios": ["Inbound API Requests","Outbound Network Requests","Page View","Plugin executions","SDK executions","Plugin executions excep","SDK executions excep"]
}
```

Power Automate:
```json
{
  "resourceProvider": "cloud flow",
  "scenarios": ["cloud flow actions","cloud flow runs","cloud flow triggers"]
}
```

Dynamics Customer Service:
```json
{
  "resourceProvider": "customer service",
  "scenarios": ["default"]
}
```

Power Apps:
```json
{
  "resourceProvider": "power apps",
  "scenarios": ["mobile sync", "UCI"]
}
```

##### API Details

API URL: `[POST] https://emea.csanalytics.powerplatform.microsoft.com/api/v2/sinks/appinsights/connections`
Sample Payload:
```json
{
   "source": "AppInsight",
   "environments": [
      {
         "environmentId": "00000000-0000-0000-0000-000000000000",
         "organizationId": "00000000-0000-0000-0000-000000000000"
      }
   ],
   "sink": {
      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroupName/providers/microsoft.insights/components/resourceName",
      "subscriptionId": "00000000-0000-0000-0000-000000000000",
      "resourceGroupName": "resourceGroupName-d-rg",
      "resourceName": "resourceName-d-appInsights",
      "key": "00000000-0000-0000-0000-000000000000"
   },
   "packageName": "package name",
   "resourceProvider": "customer service",
   "scenarios": [
      "default"
   ]
}
```

--------------------------------

#### pac analytics export-app-insights delete

Usage: `pac analytics export-app-insights delete [--package-name]`

|command|description|
|--|--|
|`---package-name`|Name of the data export. (alias: -pn)|

##### API Details

API URL: `[DELETE] https://emea.csanalytics.powerplatform.microsoft.com/api/v2/sinks/AppInsights/sources/AppInsight/connection/[environmentId]`

--------------------------------

#### pac analytics export-app-insights show

> This option isn't available in the Power Platform Admin Center, making it difficult to verify the selected export settings.

Usage: `pac analytics export-app-insights show [--package-name] [--json]`

|command|description|
|--|--|
|`--package-name`|Name of the data export. (alias: -pn)|
|`--json`|Returns the output of the command as a JSON formatted string.|

--------------------------------

Further deep dives are encouraged.

**Comment by Toftager** [comment]
Did this ever gain any traction? We are running into the same requirement

/Peter Toftager-Larsen

**Comment by sachyyn** [comment]
Is there now a supported app-only API for creating and verifying environment-level Copilot Studio telemetry exports to an existing Application Insights resource?

In a test tenant, a separately registered Power Platform management application can enumerate environments and obtain Dataverse application-user access with certificate authentication. That part works without a signed-in administrator. The remaining requirement is to configure the environment export automatically when another supported environment is added.

Microsoft's Terraform provider still documents service-principal authentication as unsupported for both the analytics_data_exports data source and the per-bot copilot_studio_application_insights resource:
- https://github.com/microsoft/terraform-provider-power-platform/blob/main/docs/data-sources/analytics_data_exports.md
- https://github.com/microsoft/terraform-provider-power-platform/blob/main/docs/resources/copilot_studio_application_insights.md

Could you confirm whether environment-level export creation/readback has a supported service-principal path, and which API and permissions to use? If not, is there a supported tenant policy or environment-group rule that automatically applies this export to future environments? We are trying to avoid recurring administrator sign-ins or per-agent setup, and do not want to treat the proposed PAC commands here as shipped functionality.
